File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        int count = 0, left = 0, right = 0, sum = 1;
        while (right < nums.size())
        {
            sum *= nums[right];
            count = (nums[right] < k ? count + 1 : count);
            while (sum >= k && left < right)
                sum /= nums[left++];
            count += right++ - left;
        }
        return (count);
    }
};

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numSubarrayProductLessThanK = function(nums, k) {
    let count = 0, left = 0, right = 0, sum = 1;
    while (right < nums.length) {
        sum = sum * nums[right];
        count += (nums[right] < k) ? 1 : 0;
        while (sum >= k && left < right)
        {
            sum = sum / nums[left];
            left++;
        }
        count += right++ - left;
    }
    return (count);
};